[2020-12-22 14:14:53,271] {taskinstance.py:655} INFO - Dependencies all met for <TaskInstance: loan_prediction_train_model.preprocess_data 2020-12-22T12:43:23.210460+00:00 [queued]>
[2020-12-22 14:14:53,286] {taskinstance.py:655} INFO - Dependencies all met for <TaskInstance: loan_prediction_train_model.preprocess_data 2020-12-22T12:43:23.210460+00:00 [queued]>
[2020-12-22 14:14:53,287] {taskinstance.py:866} INFO - 
--------------------------------------------------------------------------------
[2020-12-22 14:14:53,287] {taskinstance.py:867} INFO - Starting attempt 43 of 48
[2020-12-22 14:14:53,287] {taskinstance.py:868} INFO - 
--------------------------------------------------------------------------------
[2020-12-22 14:14:53,300] {taskinstance.py:887} INFO - Executing <Task(LivyBatchOperator): preprocess_data> on 2020-12-22T12:43:23.210460+00:00
[2020-12-22 14:14:53,303] {standard_task_runner.py:53} INFO - Started process 59629 to run task
[2020-12-22 14:14:53,366] {logging_mixin.py:112} INFO - Running %s on host %s <TaskInstance: loan_prediction_train_model.preprocess_data 2020-12-22T12:43:23.210460+00:00 [running]> dcb43f18a24b
[2020-12-22 14:14:53,392] {logging_mixin.py:112} INFO - [2020-12-22 14:14:53,391] {batch.py:214} INFO - Submitting the batch to Livy... Payload:
{
  "file": "s3a://alanchn31-loan-default-prediction/dist/main.py",
  "args": [
    "--job",
    "preprocess_data",
    "--phase",
    "train",
    "--mode",
    "aws",
    "--awsKey",
    "AKIAUW4FKHTMR2UR3INH",
    "--awsSecretKey",
    "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",
    "--s3Bucket",
    "alanchn31-loan-default-prediction"
  ],
  "pyFiles": [
    "s3a://alanchn31-loan-default-prediction/dist/src.zip"
  ],
  "name": "preprocess_data_scheduled__2020-12-22T12:43:23.210460+00:00"
}
[2020-12-22 14:14:53,401] {logging_mixin.py:112} INFO - [2020-12-22 14:14:53,399] {base_hook.py:84} INFO - Using connection to: id: livy. Host: http://ec2-3-236-26-7.compute-1.amazonaws.com, Port: 8998, Schema: None, Login: None, Password: None, extra: None
[2020-12-22 14:14:53,403] {logging_mixin.py:112} INFO - [2020-12-22 14:14:53,403] {http_hook.py:136} INFO - Sending 'POST' to url: http://ec2-3-236-26-7.compute-1.amazonaws.com:8998/batches
[2020-12-22 14:14:53,421] {logging_mixin.py:112} INFO - [2020-12-22 14:14:53,420] {batch.py:166} INFO - Batch successfully submitted with id = 39.
[2020-12-22 14:14:53,421] {logging_mixin.py:112} INFO - [2020-12-22 14:14:53,421] {batch.py:80} INFO - Getting batch 39 status...
[2020-12-22 14:14:53,432] {logging_mixin.py:112} INFO - [2020-12-22 14:14:53,432] {base_hook.py:84} INFO - Using connection to: id: livy. Host: http://ec2-3-236-26-7.compute-1.amazonaws.com, Port: 8998, Schema: None, Login: None, Password: None, extra: None
[2020-12-22 14:14:53,433] {logging_mixin.py:112} INFO - [2020-12-22 14:14:53,433] {http_hook.py:136} INFO - Sending 'GET' to url: http://ec2-3-236-26-7.compute-1.amazonaws.com:8998/batches/39
[2020-12-22 14:14:53,446] {logging_mixin.py:112} INFO - [2020-12-22 14:14:53,446] {batch.py:90} INFO - Batch 39 has not finished yet (state is 'starting')
[2020-12-22 14:15:13,466] {logging_mixin.py:112} INFO - [2020-12-22 14:15:13,466] {batch.py:80} INFO - Getting batch 39 status...
[2020-12-22 14:15:13,473] {logging_mixin.py:112} INFO - [2020-12-22 14:15:13,473] {base_hook.py:84} INFO - Using connection to: id: livy. Host: http://ec2-3-236-26-7.compute-1.amazonaws.com, Port: 8998, Schema: None, Login: None, Password: None, extra: None
[2020-12-22 14:15:13,475] {logging_mixin.py:112} INFO - [2020-12-22 14:15:13,475] {http_hook.py:136} INFO - Sending 'GET' to url: http://ec2-3-236-26-7.compute-1.amazonaws.com:8998/batches/39
[2020-12-22 14:15:13,483] {logging_mixin.py:112} INFO - [2020-12-22 14:15:13,482] {batch.py:299} INFO - --------------------------------------------------Full log for batch 39--------------------------------------------------
[2020-12-22 14:15:13,490] {logging_mixin.py:112} INFO - [2020-12-22 14:15:13,489] {base_hook.py:84} INFO - Using connection to: id: livy. Host: http://ec2-3-236-26-7.compute-1.amazonaws.com, Port: 8998, Schema: None, Login: None, Password: None, extra: None
[2020-12-22 14:15:13,493] {logging_mixin.py:112} INFO - [2020-12-22 14:15:13,493] {http_hook.py:136} INFO - Sending 'GET' to url: http://ec2-3-236-26-7.compute-1.amazonaws.com:8998/batches/39/log?from=0&size=100
[2020-12-22 14:15:13,500] {logging_mixin.py:112} INFO - [2020-12-22 14:15:13,500] {batch.py:309} INFO - stdout: 
[2020-12-22 14:15:13,500] {logging_mixin.py:112} INFO - [2020-12-22 14:15:13,500] {batch.py:309} INFO - 2020-12-22 14:14:55,740 WARN util.NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2020-12-22 14:15:13,500] {logging_mixin.py:112} INFO - [2020-12-22 14:15:13,500] {batch.py:309} INFO - Traceback (most recent call last):
[2020-12-22 14:15:13,500] {logging_mixin.py:112} INFO - [2020-12-22 14:15:13,500] {batch.py:309} INFO -   File "/tmp/spark-94b505b9-192f-41b3-8901-5aa13deb0d1a/main.py", line 87, in <module>
[2020-12-22 14:15:13,501] {logging_mixin.py:112} INFO - [2020-12-22 14:15:13,501] {batch.py:309} INFO -     main()
[2020-12-22 14:15:13,501] {logging_mixin.py:112} INFO - [2020-12-22 14:15:13,501] {batch.py:309} INFO -   File "/tmp/spark-94b505b9-192f-41b3-8901-5aa13deb0d1a/main.py", line 59, in main
[2020-12-22 14:15:13,501] {logging_mixin.py:112} INFO - [2020-12-22 14:15:13,501] {batch.py:309} INFO -     for filename in listdir(ROOT_DIR)
[2020-12-22 14:15:13,501] {logging_mixin.py:112} INFO - [2020-12-22 14:15:13,501] {batch.py:309} INFO - NameError: name 'listdir' is not defined
[2020-12-22 14:15:13,501] {logging_mixin.py:112} INFO - [2020-12-22 14:15:13,501] {batch.py:309} INFO - 2020-12-22 14:15:00,406 INFO util.ShutdownHookManager: Shutdown hook called
[2020-12-22 14:15:13,502] {logging_mixin.py:112} INFO - [2020-12-22 14:15:13,501] {batch.py:309} INFO - 2020-12-22 14:15:00,407 INFO util.ShutdownHookManager: Deleting directory /tmp/spark-94b505b9-192f-41b3-8901-5aa13deb0d1a
[2020-12-22 14:15:13,502] {logging_mixin.py:112} INFO - [2020-12-22 14:15:13,502] {batch.py:309} INFO - 2020-12-22 14:15:00,413 INFO impl.MetricsSystemImpl: Stopping s3a-file-system metrics system...
[2020-12-22 14:15:13,502] {logging_mixin.py:112} INFO - [2020-12-22 14:15:13,502] {batch.py:309} INFO - 2020-12-22 14:15:00,413 INFO impl.MetricsSystemImpl: s3a-file-system metrics system stopped.
[2020-12-22 14:15:13,502] {logging_mixin.py:112} INFO - [2020-12-22 14:15:13,502] {batch.py:309} INFO - 2020-12-22 14:15:00,413 INFO impl.MetricsSystemImpl: s3a-file-system metrics system shutdown complete.
[2020-12-22 14:15:13,502] {logging_mixin.py:112} INFO - [2020-12-22 14:15:13,502] {batch.py:309} INFO - 
stderr: 
[2020-12-22 14:15:13,503] {logging_mixin.py:112} INFO - [2020-12-22 14:15:13,502] {batch.py:309} INFO - 
YARN Diagnostics: 
[2020-12-22 14:15:13,503] {logging_mixin.py:112} INFO - [2020-12-22 14:15:13,503] {batch.py:309} INFO - spark-submit start failed
[2020-12-22 14:15:13,503] {logging_mixin.py:112} INFO - [2020-12-22 14:15:13,503] {batch.py:318} INFO - --------------------------------------------------End of full log for batch 39--------------------------------------------------
[2020-12-22 14:15:13,503] {logging_mixin.py:112} INFO - [2020-12-22 14:15:13,503] {batch.py:335} INFO - Closing batch with id = 39
[2020-12-22 14:15:13,510] {logging_mixin.py:112} INFO - [2020-12-22 14:15:13,510] {base_hook.py:84} INFO - Using connection to: id: livy. Host: http://ec2-3-236-26-7.compute-1.amazonaws.com, Port: 8998, Schema: None, Login: None, Password: None, extra: None
[2020-12-22 14:15:13,511] {logging_mixin.py:112} INFO - [2020-12-22 14:15:13,511] {http_hook.py:136} INFO - Sending 'DELETE' to url: http://ec2-3-236-26-7.compute-1.amazonaws.com:8998/batches/39
[2020-12-22 14:15:13,518] {logging_mixin.py:112} INFO - [2020-12-22 14:15:13,518] {batch.py:340} INFO - Batch 39 has been closed
[2020-12-22 14:15:13,527] {taskinstance.py:1128} ERROR - Batch 39 failed with state 'dead'
Traceback (most recent call last):
  File "/opt/bitnami/python/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 966, in _run_raw_task
    result = task_copy.execute(context=context)
  File "/usr/local/airflow/plugins/airflow_livy/batch.py", line 173, in execute
    ).execute(context)
  File "/opt/bitnami/python/lib/python3.6/site-packages/airflow/sensors/base_sensor_operator.py", line 107, in execute
    while not self.poke(context):
  File "/usr/local/airflow/plugins/airflow_livy/batch.py", line 96, in poke
    raise AirflowException(f"Batch {self.batch_id} failed with state '{state}'")
airflow.exceptions.AirflowException: Batch 39 failed with state 'dead'
[2020-12-22 14:15:13,529] {taskinstance.py:1151} INFO - Marking task as UP_FOR_RETRY
[2020-12-22 14:15:18,318] {logging_mixin.py:112} INFO - [2020-12-22 14:15:18,317] {local_task_job.py:103} INFO - Task exited with return code 1
